pkg/tlsutil/tls.go

```diff
@@ -111,3 +119,135 @@ type CertGenerator interface {
 	//   ca.key: ..
 	GenerateCert(cr runtime.Object, service *v1.Service, config *CertConfig) (*v1.Secret, *v1.ConfigMap, *v1.Secret, error)
 }
+
+const (
+	// TLSPrivateCAKeyKey is the key for the private CA key field.
+	TLSPrivateCAKeyKey = "ca.key"
+	// TLSCertKey is the key for tls CA certificates.
+	TLSCACertKey = "ca.crt"
+)
+
+type SDKCertGenerator struct {
+	// appKeyAndCertMap holds the application key and cert for a given cr + config.CertName.
+	appKeyAndCertMap sync.Map
+	// caKeyAndCertMap holds the CA key and cert for a given cr.
+	caKeyAndCertMap sync.Map
```

Comment:
Why is a `sync.Map` needed here? I might not understand the interaction model so I am more asking for my understanding.

Reply:
I was thinking about making this function thread safe. On a second thought, sync.map is probably not enough. I'll just use a lock instead and use normal map.